sys/linux/devio.txt

```diff
@@ -0,0 +1,129 @@
+# Copyright 2019 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+# https://elixir.bootlin.com/linux/latest/source/drivers/usb/core/devio.c
+
+include <linux/ioctl.h>
+include <linux/usbdevice_fs.h>
+include <uapi/linux/fcntl.h>
+
+define MAXDRIVERNAME	USBDEVFS_MAXDRIVERNAME + 1
+
+resource fd_devio[fd]
+
+syz_open_dev$usb(dev ptr[in, string["/dev/bus/usb/00#/00#"]], id intptr, flags flags[open_flags]) fd_devio
+
+read$devio(fd fd_devio, buf buffer[out], count len[buf])
+
+mmap$devio(addr vma, len len[addr], prot flags[mmap_prot], flags flags[mmap_flags], fd fd_devio, offset fileoff)
+_ = __NR_mmap2
+
+ioctl$USBDEVFS_REAPURB(fd fd_devio, cmd const[USBDEVFS_REAPURB], arg ptr[out, int64])
+ioctl$USBDEVFS_REAPURBNDELAY(fd fd_devio, cmd const[USBDEVFS_REAPURBNDELAY], arg ptr[out, int64])
+ioctl$USBDEVFS_CONTROL(fd fd_devio, cmd const[USBDEVFS_CONTROL], arg ptr[in, usbdevfs_ctrltranfer])
+ioctl$USBDEVFS_BULK(fd fd_devio, cmd const[USBDEVFS_BULK], arg ptr[in, usbdevfs_bulktransfer])
+ioctl$USBDEVFS_RESETEP(fd fd_devio, cmd const[USBDEVFS_RESETEP], arg ptr[in, int32])
+ioctl$USBDEVFS_RESET(fd fd_devio, cmd const[USBDEVFS_RESET])
+ioctl$USBDEVFS_CLEAR_HALT(fd fd_devio, cmd const[USBDEVFS_CLEAR_HALT], arg ptr[in, int32])
+ioctl$USBDEVFS_GETDRIVER(fd fd_devio, cmd const[USBDEVFS_GETDRIVER], arg ptr[inout, usbdevfs_getdriver])
+ioctl$USBDEVFS_CONNECTINFO(fd fd_devio, cmd const[USBDEVFS_CONNECTINFO], arg ptr[out, usbdevfs_connectinfo])
+ioctl$USBDEVFS_SETINTERFACE(fd fd_devio, cmd const[USBDEVFS_SETINTERFACE], arg ptr[in, usbdevfs_setinterface])
+ioctl$USBDEVFS_SETCONFIGURATION(fd fd_devio, cmd const[USBDEVFS_SETCONFIGURATION], arg ptr[in, int32])
+ioctl$USBDEVFS_SUBMITURB(fd fd_devio, cmd const[USBDEVFS_SUBMITURB], arg ptr[in, usbdevfs_urb])
+ioctl$USBDEVFS_DISCARDURB(fd fd_devio, cmd const[USBDEVFS_DISCARDURB], arg ptr[in, int64])
```

Comment:
Add a comment that this accepts a pointer to a previously submitted URB in `arg`.